src/index.js

```diff
@@ -0,0 +1,42 @@
+const fsp = require('fs').promises;
+const { readDataFiles } = require('./utils/data-parser');
+const { MatterParser } = require('./utils/matter-parser');
+
+async function getFileTimes(fileName, addCreated, addModified) {
+  const result = {};
+  if (!addCreated && !addModified) {
+    return result;
+  }
+  const stats = await fsp.stat(fileName);
+  if (addCreated) {
+    result.created = stats.birthtime.toISOString();
+  }
+  if (addModified) {
+    result.modified = stats.mtime.toISOString();
+  }
+  return result;
+}
+
+async function processFrontmatterFiles({
+  inputFilePatterns,
+  dataFilePatterns,
+  data = {},
+  addCreated = false,
+  addModified = false,
+}) {
+  const dataContents = await readDataFiles(dataFilePatterns);
+  const fileData = Object.assign.apply(this, [{}, ...dataContents]);
```

Comment:
I'm a little confused seeing this pattern. What's thidms expected to be? The module? Something else?

Reply:
it merges an array of objects into one new object, like `[{a:1}, {b:2}]` into `[{a:1, b:2}]`